src/lib/addEventListeners.spec.js

```diff
@@ -1,19 +1,221 @@
+
+      const dialogState = { wrapperId: wrapper.id, wrapper, prev: sinon.spy() };
+      addEventListeners(dialogState);
+      button.click();
+      expect(dialogState.prev).to.have.been.called();
+
+      // Cleanup
+      wrapper.remove();
+    });
   });
-  it('adds next/prev event listeners to buttons found via attributes');
 
-  it('adds onKeyDown listener to the wrapper for keyCode 27 if element is visible');
+  describe('Keydown events on the wrapper', () => {
+    it('adds keydown listener to the wrapper', () => {
+      const wrapper = document.createElement('div');
+      wrapper.id = 'test';
+      wrapper.addEventListener = sinon.spy();
+
+      document.body.appendChild(wrapper);
+
+      const dialogState = { wrapperId: wrapper.id, wrapper };
+
+      addEventListeners(dialogState);
+      simulant.fire(wrapper, 'keydown');
+
+      // First call adds the 'hide' 'click' handler,
+      // second call is the relevant one.
+      // This is a little brittle. Is there no better,
+      // less execution-order dependent way to do this?
+      expect(wrapper.addEventListener.getCall(1).args[0]).to.equal('keydown');
+
+      // Cleanup
+      wrapper.remove();
+    });
+
+    it('executes hide() on keydown when keyCode is 27 & wrapper is visible', () => {
+      const wrapper = document.createElement('div');
+      wrapper.id = 'test';
+
+      document.body.appendChild(wrapper);
+
+      const dialogState = { wrapperId: wrapper.id, wrapper, isVisible: true, hide: sinon.spy() };
+
+      addEventListeners(dialogState);
+      simulant.fire(wrapper, 'keydown', { keyCode: 27 });
+
+      expect(dialogState.hide).to.have.been.called();
+
+      // Cleanup
+      wrapper.remove();
+    });
+
+    it('Does not execute hide() on keydown when keyCode is 27 & wrapper is *NOT* visible', () => {
+      const wrapper = document.createElement('div');
+      wrapper.id = 'test';
+
+      document.body.appendChild(wrapper);
+
+      const dialogState = { wrapperId: wrapper.id, wrapper, isVisible: false, hide: sinon.spy() };
+
+      addEventListeners(dialogState);
+      simulant.fire(wrapper, 'keydown', { keyCode: 27 });
+
+      expect(dialogState.hide).to.not.have.been.called();
+
+      // Cleanup
+      wrapper.remove();
+    });
+
+    it('executes keepFocus on keydown when keyCode is 9 & wrapper is visible', () => {
+      const wrapper = document.createElement('div');
+      wrapper.id = 'test';
+
+      document.body.appendChild(wrapper);
+
+      const dialogState = { wrapperId: wrapper.id, wrapper, isVisible: true };
+      const keepFocusSpy = sinon.spy();
+
+      rewire('keepFocus', keepFocusSpy);
+
+      addEventListeners(dialogState);
+      simulant.fire(wrapper, 'keydown', { keyCode: 9 });
+
+      expect(keepFocusSpy).to.have.been.called();
+
+      // Cleanup
+      wrapper.remove();
+    });
+
+    it('Does not execute keepFocus on keydown when keyCode is 9 & wrapper is *NOT* visible', () => {
```

Comment:
for the sake of DRYness we can make these assertions within the test above using the same setup and cleanup or abstract these and only change the keyCode and assertion